packages/dapp/components/lib/hooks/useWeb3.tsx

```diff
@@ -47,7 +47,17 @@ export const UseWeb3Provider: FC<ChildrenShim> = ({ children }) => {
 };
 
 const useWeb3 = (): [Web3State] => {
-  const provider = useProvider();
+  let provider: PossibleProviders = null;
+  if (typeof window !== "undefined") {
+    if (window.ethereum) {
+      provider = new Web3Provider(window.ethereum);
+    } else if (IS_DEV) {
+      provider = new JsonRpcProvider(LOCAL_NODE_ADDRESS);
+    } else {
+      provider = new WebSocketProvider("wss://ethereum-rpc.publicnode.com", 1);
```

Comment:
I remember we had issues with the amount of requests we were making for the audit app. Maybe websockets are the solution.